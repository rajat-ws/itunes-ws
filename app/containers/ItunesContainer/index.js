import React from 'react';

export default function ItunesContainer() {
  return <div> itunesContainer page </div>;
}
