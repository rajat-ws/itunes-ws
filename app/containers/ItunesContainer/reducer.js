/*
 *
 * itunesContainer reducer
 *
 */

import produce from 'immer';
import { createActions } from 'reduxsauce';

export const initialState = {
  searchTerm: null,
  gridData: {},
  searchError: null
};

export const { Types: itunesContainerTypes, Creators: itunesContainerCreators } = createActions({
  searchItunes: ['searchTerm'],
  successSearchItunes: ['data'],
  failureSearchItunes: ['error'],
  clearGridData: {}
});

/* eslint-disable default-case, no-param-reassign */
export const itunesContainerReducer = (state = initialState, action) =>
  produce(state, (draft) => {
    switch (action.type) {
      case itunesContainerTypes.SEARCH_ITUNES:
        draft.searchTerm = action.searchTerm;
        break;

      case itunesContainerTypes.SUCCESS_SEARCH_ITUNES:
        draft.searchError = null;
        draft.gridData = action.data;
        break;

      case itunesContainerTypes.FAILURE_SEARCH_ITUNES:
        draft.searchError = action.error;
        draft.searchTerm = null;
        draft.gridData = {};
        break;

      case itunesContainerTypes.CLEAR_GRID_DATA:
        draft.gridData = {};
        draft.searchTerm = null;
        break;
    }
  });

export default itunesContainerReducer;
