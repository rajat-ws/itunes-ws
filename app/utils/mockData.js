export const MOCK_TRACK_DATA = {
  wrapperType: 'track',
  kind: 'song',
  artistId: 76531581,
  collectionId: 1068455603,
  trackId: 1068455608,
  artistName: 'Tom MacDonald',
  collectionName: 'See You Tomorrow',
  trackName: 'Greenday',
  collectionCensoredName: 'See You Tomorrow',
  trackCensoredName: 'Greenday',
  artistViewUrl: 'https://music.apple.com/us/artist/tom-macdonald/76531581?uo=4',
  collectionViewUrl: 'https://music.apple.com/us/album/greenday/1068455603?i=1068455608&uo=4',
  trackViewUrl: 'https://music.apple.com/us/album/greenday/1068455603?i=1068455608&uo=4',
  previewUrl:
    'https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview125/v4/51/bb/61/51bb6190-6865-3050-af01-0f64502bc376/mzaf_14313183610807291436.plus.aac.p.m4a',
  artworkUrl30:
    'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/dd/64/78/dd6478bc-e684-1b69-2e4e-c7e1d6a36c0a/045635953260.jpg/30x30bb.jpg',
  artworkUrl60:
    'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/dd/64/78/dd6478bc-e684-1b69-2e4e-c7e1d6a36c0a/045635953260.jpg/60x60bb.jpg',
  artworkUrl100:
    'https://is3-ssl.mzstatic.com/image/thumb/Music124/v4/dd/64/78/dd6478bc-e684-1b69-2e4e-c7e1d6a36c0a/045635953260.jpg/100x100bb.jpg',
  collectionPrice: 9.99,
  trackPrice: 0.99,
  releaseDate: '2015-12-21T12:00:00Z',
  collectionExplicitness: 'explicit',
  trackExplicitness: 'explicit',
  discCount: 1,
  discNumber: 1,
  trackCount: 17,
  trackNumber: 3,
  trackTimeMillis: 256520,
  country: 'USA',
  currency: 'USD',
  primaryGenreName: 'Hip-Hop/Rap',
  contentAdvisoryRating: 'Explicit',
  isStreamable: true
};
