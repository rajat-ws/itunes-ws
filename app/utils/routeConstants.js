export default {
  itunes: {
    route: '/',
    props: {
      maxWidth: 32,
      padding: 20
    },
    exact: true
  },
  trackDetails: {
    route: '/tracks/:trackId',
    exact: true
  }
};
