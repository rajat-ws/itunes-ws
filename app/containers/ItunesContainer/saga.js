import { call, put, takeLatest } from 'redux-saga/effects';
import { getSongs } from '@app/services/itunesApi';
import { itunesContainerTypes, itunesContainerCreators } from './reducer';

const { SEARCH_ITUNES } = itunesContainerTypes;
const { successSearchItunes, failureSearchItunes } = itunesContainerCreators;

export function* requestSearchItunes(action) {
  const res = yield call(getSongs, action.searchTerm);
  const { data, ok } = res;

  //check if ok
  if (ok) {
    yield put(successSearchItunes(data));
  } else {
    yield put(failureSearchItunes(data));
  }
}

//exporting itunesContainer saga
export default function* itunesContainerSaga() {
  yield takeLatest(SEARCH_ITUNES, requestSearchItunes);
}
