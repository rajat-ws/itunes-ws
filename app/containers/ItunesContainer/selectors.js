import { createSelector } from 'reselect';
import get from 'lodash/get';
import { initialState } from './reducer';

/**
 * Direct selector to the itunesContainer state domain
 */

export const selectItunesContainerDomain = (state) => state.itunesContainer || initialState;

export const selectItunesContainer = () => createSelector(selectItunesContainerDomain, (substate) => substate);

export const selectSearchTerm = () =>
  createSelector(selectItunesContainerDomain, (substate) => get(substate, 'searchTerm'));

export const selectSearchError = () =>
  createSelector(selectItunesContainerDomain, (substate) => get(substate, 'searchError'));

export const selectGridData = () =>
  createSelector(selectItunesContainerDomain, (substate) => get(substate, 'gridData'));
