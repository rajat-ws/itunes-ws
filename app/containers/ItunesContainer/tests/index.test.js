/*
 ** Tests for ItunesContainer
 */
import React from 'react';
import { renderProvider } from '@app/utils/testUtils';
import ItunesContainer from '../index';

describe('ItunesContainer Tests', () => {
  it('should render and match to the snapshot', () => {
    const { baseElement } = renderProvider(<ItunesContainer />);
    expect(baseElement).toMatchSnapshot();
  });
});
