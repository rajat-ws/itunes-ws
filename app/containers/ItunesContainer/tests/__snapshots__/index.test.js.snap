// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItunesContainer Tests should render and match to the snapshot 1`] = `
<body>
  <div>
    <div>
       itunesContainer page 
    </div>
  </div>
</body>
`;
